`TestNewFromFlags` test will fail when `UP_ACCOUNT` is set

<!--
Thank you for helping to improve Crossplane!

Please be sure to search for open issues before raising a new one. We use issues
for bug reports and feature requests. Please find us at https://slack.crossplane.io
for questions, support, and discussion.
-->

### What happened?
<!--
Please let us know what behaviour you expected and how Crossplane diverged from
that behaviour.
-->
When running `make reviewable`, if environment variable `UP_ACCOUNT` is set then it will fail the `TestNewFromFlags` test.

### How can we reproduce it?
<!--
Help us to reproduce your bug as succinctly and precisely as possible. Artifacts
such as example manifests or a script that triggers the issue are highly
appreciated!
-->

```bash
// Simulate UP_ACCOUNT is set in the environment variable
UP_ACCOUNT=my-account make reviewable
```
Here is the result from the above command.

```
--- FAIL: TestNewFromFlags (0.00s)
    --- FAIL: TestNewFromFlags/NoPreExistingProfile (0.00s)
        context_test.go:297:
            We should successfully return a Context if a pre-existing profile does not exist.
            NewFromFlags(...): -want error, +got error:
              &upbound.Context{
              	ProfileName:           "",
              	Profile:               {},
              	Token:                 "",
            - 	Account:               "",
            + 	Account:               "my-account",
              	Domain:                &{Scheme: "https", Host: "upbound.io"},
              	InsecureSkipTLSVerify: false,
              	... // 5 ignored and 2 identical fields
              }
    --- FAIL: TestNewFromFlags/SuppliedNotExistAllowEmpty (0.00s)
        context_test.go:297:
            We should successfully return a Context if a supplied profile does not exist and.
            NewFromFlags(...): -want error, +got error:
              &upbound.Context{
              	ProfileName:           "not-here",
              	Profile:               {},
              	Token:                 "",
            - 	Account:               "",
            + 	Account:               "my-account",
              	Domain:                &{Scheme: "https", Host: "upbound.io"},
              	InsecureSkipTLSVerify: false,
              	... // 5 ignored and 2 identical fields
              }
    --- FAIL: TestNewFromFlags/PreExistingProfileNoBaseConfig (0.00s)
        context_test.go:297:
            We should successfully return a Context if a pre-existing profile exists, but does not have a base config
            NewFromFlags(...): -want error, +got error:
              &upbound.Context{
              	ProfileName:           "default",
              	Profile:               {ID: "someone@upbound.io", Type: "user", Session: "a token"},
              	Token:                 "",
            - 	Account:               "",
            + 	Account:               "my-account",
              	Domain:                &{Scheme: "https", Host: "upbound.io"},
              	InsecureSkipTLSVerify: false,
              	... // 5 ignored and 2 identical fields
              }
    --- FAIL: TestNewFromFlags/PreExistingProfileBaseConfigSetProfile (0.00s)
        context_test.go:297:
            We should return a Context that includes the persisted Profile from base config
            NewFromFlags(...): -want error, +got error:
              &upbound.Context{
              	ProfileName:           "default",
              	Profile:               {ID: "someone@upbound.io", Type: "user", Session: "a token", BaseConfig: {"UP_ACCOUNT": "my-org", "UP_DOMAIN": "https://local.upbound.io", "UP_INSECURE_SKIP_TLS_VERIFY": "true"}},
              	Token:                 "",
            - 	Account:               "my-org",
            + 	Account:               "my-account",
              	Domain:                &{Scheme: "https", Host: "local.upbound.io"},
              	InsecureSkipTLSVerify: true,
              	... // 5 ignored and 2 identical fields
              }
FAIL
```


### What environment did it happen in?
Crossplane version: `master` branch

<!--
Include at least the version or commit of Crossplane you were running. Consider
also including your:

* Cloud provider or hardware configuration
* Kubernetes version (use `kubectl version`)
* Kubernetes distribution (e.g. Tectonic, GKE, OpenShift)
* OS (e.g. from /etc/os-release)
* Kernel (e.g. `uname -a`)
-->

--- a/internal/xpkg/upbound/context_test.go
+++ b/internal/xpkg/upbound/context_test.go
@@ -19,6 +19,7 @@ package upbound
 import (
 	"fmt"
 	"net/url"
+	"os"
 	"path/filepath"
 	"testing"
 
@@ -274,6 +275,11 @@ func TestNewFromFlags(t *testing.T) {
 	}
 
 	for name, tc := range cases {
+		// Unset common UP env vars used by the test to avoid unexpect behaviours describe in #5721
+		os.Unsetenv("UP_ACCOUNT")
+		os.Unsetenv("UP_DOMAIN")
+		os.Unsetenv("UP_PROFILE")
+		os.Unsetenv("UP_INSECURE_SKIP_TLS_VERIFY")
 		t.Run(name, func(t *testing.T) {
 			flags := Flags{}
 			parser, _ := kong.New(&flags)

